File: cpp/calc.cpp
#include <iostream>
#include "mpreal.h"

using namespace mpfr;

mpfr::mpreal fact(int n) {
	return (n == 0 || n == 1) ? 1 : fact(n - 1) * n;
}

mpfr::mpreal calcit(int maxiter) {

	mpfr::mpreal s = mpfr::mpreal(0);

	for (int n = 0; n < maxiter; ++n) {
		s += mpfr::pow(-1, n) * fact(6 * n) * (13591409 + n * mpfr::mpreal(545140134)) /
             mpfr::pow(fact(n), 3) / fact(3 * n) /
             mpfr::pow(100100025 * 8 * mpfr::mpreal(327843840), n);
	}

	return 53360 * mpfr::pow(640320, mpfr::mpreal("0.5")) / s; 

}

int main(int argc, char *argv[]) {

	const int prec = 7;

	mpfr::mpreal::set_default_prec(50 * 50 * prec);
	std::cout.precision(50 * 75 * prec);

	std::cout << calcit(50 * prec) << std::endl << mpfr::mpreal::get_default_prec();

    return 0;
}

File: rb/calc.rb
require 'bigdecimal'

class Integer
	def fact
		f = 1
		for i in 1..self
			f *= i
		end
		f
  	end
end

prec = 3
BigDecimal::mode(BigDecimal::ROUND_MODE, BigDecimal::ROUND_HALF_UP)
@decprec = 2500 * prec

def calcit(maxiter)
	s = BigDecimal.new('0', @decprec)
	
	maxiter.times do |n|
		s += (-1) ** n * (6 * n).fact * (13591409 + n * BigDecimal.new(545140134, @decprec)) /
			n.fact ** 3 / (3 * n).fact / (100100025 * 8 * BigDecimal.new(327843840, @decprec)) ** n
	end

	return 53360 * BigDecimal.new(640320, @decprec) ** BigDecimal.new('0.5', @decprec) / s
end

puts(calcit(50 * prec).to_s("#{3750 * prec}F"))
